httpx/_multipart.py

```diff
@@ -214,16 +206,18 @@ def __init__(
     def _iter_fields(
         self, data: RequestData, files: RequestFiles
     ) -> typing.Iterator[typing.Union[FileField, DataField]]:
-        for name, value in data.items():
-            if isinstance(value, (tuple, list)):
-                for item in value:
-                    yield DataField(name=name, value=item)
+        for name, data_value in data.items():
+            if data_value is None or isinstance(
+                data_value, (*PRIMITIVE_DATA_TYPES, bytes)
```

Comment:
DataField accepts `bytes` in addition to the other "primitive types". I'm not sure where else `PrimitiveTypes` is being used but we might want to consider expanding everywhere to accept bytes instead of just here.